### ISSUE TITLE
Rule Change: add `uniqueItems: true` in `new-cap`, `no-shadow`, `no-underscore-dangle`, `no-warning-comments`, and `class-methods-use-this` in v11

### ISSUE THREAD

**Issue by lumirlumir** [eslint/eslint#20334]
## What rule do you want to change?

`new-cap`, `no-shadow`, `no-underscore-dangle`, `no-warning-comments`, and `class-methods-use-this`

## What change do you want to make?

Generate more warnings

## How do you think the change should be implemented?

A new default behavior

### `no-underscore-dangle`

Add `uniqueItems: true` to the schema for the `allow` option of the `no-underscore-dangle` rule.

Given the rule's description and implementation, preventing duplicate `allow` seems valid and safe.

- https://eslint.org/docs/latest/rules/no-underscore-dangle#allow

<img width="573" height="44" alt="image" src="https://github.com/user-attachments/assets/2cf1b04a-ab7f-49d3-bc8b-c61ac032b379" />

Currently, adding the following test case to the `valid` tests does not produce any errors:

```js
{ code: "var _foo = 1", options: [{ allow: ["_foo", "_foo"] }] },
```

### `no-warning-comments`

Add `uniqueItems: true` to the schema for the `terms` option of the `no-warning-comments` rule.

Given the rule's description and implementation, preventing duplicate terms seems valid and safe.

- https://eslint.org/docs/latest/rules/no-warning-comments#terms-and-location

<img width="788" height="93" alt="image" src="https://github.com/user-attachments/assets/60883393-0ad0-4054-8739-d1630099efcb" />

Currently, adding the following test case to the `valid` tests does not produce any errors:

```js
{ code: "// any comment", options: [{ terms: ["fixme", "fixme"] }] }
```

### `class-methods-use-this`

Add `uniqueItems: true` to the schema for the `exceptMethods` option of the `class-methods-use-this` rule.

Given the rule's description and implementation, preventing duplicate exception names seems valid and safe.

- https://eslint.org/docs/latest/rules/class-methods-use-this#exceptmethods

<img width="856" height="306" alt="image" src="https://github.com/user-attachments/assets/62b442ec-79b0-430e-b0af-926dd3ae7024" />

Currently, adding the following test case to the `valid` tests does not produce any errors:

```js
		{
			code: "class A { foo() {this} bar() {} }",
			options: [{ exceptMethods: ["bar", "bar"] }], // Duplicated.
			languageOptions: { ecmaVersion: 6 },
		},
```

## What does the rule currently do for this code?

Working in progress.

## What will the rule do after it's changed?

Working in progress.

## Participation

- [x] I am willing to submit a pull request to implement this change.

## Additional comments

References:

- https://github.com/eslint/eslint/pull/20309
- https://github.com/eslint/eslint/pull/20308
- https://github.com/eslint/eslint/pull/20307
- https://github.com/eslint/eslint/pull/20306

**Comment by mdjermanovic** [comment]
Added to v11.0.0 - Need Discussion, as this is a breaking change and the scope for v10 is already frozen.
